src/modules/System/Service.php

```diff
@@ -379,22 +379,17 @@ public function getEnv($ip)
 
     public function getCurrentUrl()
     {
-        $request = $this->di['request'];
-        $pageURL = 'http';
-        $https = $request->getScheme();
-        if (isset($https) && 'https' == $https) {
-            $pageURL .= 's';
-        }
-        $pageURL .= '://';
+        $pageScheme = $_SERVER['HTTPS'] ? 'https' : 'http';
+        $pageURL = $pageScheme . '://';
 
-        $serverPort = $request->getServer('SERVER_PORT');
+        $serverPort = $_SERVER['SERVER_PORT'] ?? null;
         if (isset($serverPort) && '80' != $serverPort && '443' != $serverPort) {
-            $pageURL .= $request->getServer('SERVER_NAME') . ':' . $request->getServer('SERVER_PORT');
+            $pageURL .= $_SERVER['SERVER_NAME'] ?? null . ':' . $serverPort;
```

Comment:
Not going to make a huge deal out of this once since it matches the original behavior and I'm not sure what the best fix will be, but `null` doesn't really seem like a suitable fallback here.

Mostly just making note of something I noticed, I don't expect this to be fixed in this PR.

Reply:
Considering this - I agree. It's a difficult one - should we pull from config?

Reply:
If there's anything I can change/improve here, let me know. Otherwise, it may be for another deeper PR. 🙂

Reply:
Perhaps. I don't even know if we *need* a fallback here, truthfully